### ISSUE TITLE
Allow Docker Secrets to be used for specifying iDRAC Credentials

### ISSUE THREAD

**Issue by brownts** [DomiStyle/docker-idrac6#20]
It's undesirable to specify the user and password credentials for the idrac as environment variables as these are easily visible on the command line, in the Docker compose file, as well as by simply inspecting the container.  It would be more desirable (and more secure) to allow the credentials to be stored as Docker secrets and then accessed by the container at run-time.

It's possible that the secrets files could be checked first for the credentials and if they didn't exist to fall back on the environment variables as this would provide backwards compatibility and allow the user to choose which way they desired to supply them in addition to providing a migration path to a more secure approach.

**Comment by DomiStyle** [comment]
Good idea. I'm using Kubernetes so secrets are still exposed via environment variables but I will take a look once I got some spare time.

In the meantime you can hide them in a separate .env file instead of your docker-compose and only allow root to access the Docker Daemon and the .env file.

**Comment by DomiStyle** [comment]
@brownts I implemented support for Docker secrets for `idrac_host`, `idrac_user`, `idrac_password` and `idrac_port`.

You can check it out in the docker-secrets branch, just pull image `domistyle/idrac6:docker-secrets` from Docker Hub. An example docker-compose is available [here](https://github.com/DomiStyle/docker-idrac6/blob/docker-secrets/docker-compose.yml).

Maybe somebody can give it a try, I have no iDRAC instances running atm.

**Comment by brownts** [comment]
@DomiStyle I tested this out and it works great!  I hadn't used Docker swarm before and it's required in order to use secrets, so that was a little bit of a stumbling block.  The changes you've made to the docker work as expected (although I only used secrets for the "user" and "password", while using the environment variable for the "host" and just defaulting the "port").  I checked this out on a Dell R210ii, R510 and R610...all worked fine.  I also used a manually created secret (via "docker secret create ...") as I didn't want them to be stored in a file, thus the reason for the "external" attribute in my configuration below.

I had to adjust my configuration to use the long syntax for the port expose portion of the configuration as it seems by default the ports are not exposed to the host machine when running in a swarm unless you use "host mode" which as far as I can tell is only available in the long syntax, and the long syntax is only available starting with version 3.2.

This is the configuration I tested this with: [idrac6_config.txt](https://github.com/DomiStyle/docker-idrac6/files/4045731/idrac6_config.txt)

Thanks for adding this capability!
